### DISCUSSION TITLE
What's the difference between "img" and "img-simple" shortcodes?

### DISCUSSION THREAD

**Discussion by johnnymatthews** [thuliteio/doks#462] [General]
Both the `img` and `img-simple` shortcodes are called out in the [Doks documentation](https://getdoks.org/docs/recipes/images/) to be used when adding images to your site. But what's the difference between the two shortcodes?

The `img-simple.html` shortcode is significantly _smaller_ than the `img.html` shortcode, but I can't tell what functions or features it's missing (my Golang knowledge isn't super great, and I'm having trouble understanding what's going on).

This isn't really a support ticket, I'd just like to know what the difference between the two shortcodes is, and when I should use one over the other.

**Comment by h-enk** [comment]
The `img-simple` shortcode is for small images (say less than 300px in width), and will (just) generate an `<img>` element (see also: [The Image Embed element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Img))

The `img.html` shortcode is for large images (say equal to and greater than 300px in width), and will generate a `figure` element (see also: [The Figure with Optional Caption element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/figure)) with different image sizes for different screen sizes (configure image sizes in the Images section of `./config/default/params.toml`) — making it truly responsive.

Note, I will be implementing an image render hook for Doks soon, so you can use markdown for images (portability!) and don't have to worry about which shortcode to pick (already implemented in Hyas: [render-image.html](https://github.com/h-enk/hyas/blob/master/layouts/_default/_markup/render-image.html)).

**Reply by johnnymatthews** [reply]
Ah ok, that makes sense. Would you appreciate a PR into `getdoks.org/docs/recipes/images` explaining what you've written here?

> implementing an image render hook for Doks soon

Whoa! That'd be super cool! It'd mean that things display nicely on the Hugo site _and_ in GitHub/GitLab/whatever.

**Reply by h-enk** [reply]
> Ah ok, that makes sense. Would you appreciate a PR into getdoks.org/docs/recipes/images explaining what you've written here?

Yes, absolutely /thanks!

> Whoa! That'd be super cool! It'd mean that things display nicely on the Hugo site and in GitHub/GitLab/whatever.

Yeah, I also like this one so much — shoutout to @timtorres for triggering this!
